packages/headless/src/features/insight-user-actions/insight-user-actions-preprocessing.test.ts

```diff
@@ -0,0 +1,1202 @@
+                  searchHub: 'in-product-help',
+                  document: {},
+                  query: 'Blaze pair with iPhone not working',
+                },
+                {
+                  actionType: 'CUSTOM',
+                  timestamp: 1648743840000,
+                  eventData: {
+                    type: 'MySpeedbit App interfaceload',
+                    value: '',
+                  },
+                  searchHub: 'in-product-help',
+                  document: {},
+                },
+              ],
+            },
+            followingSessions: [
+              {
+                start: 1648825140000,
+                end: 1648826040000,
+                actions: [
+                  {
+                    actionType: 'VIEW',
+                    timestamp: 1648826040000,
+                    eventData: {},
+                    searchHub: 'in-product-help',
+                    document: {},
+                  },
+                  {
+                    actionType: 'CLICK',
+                    timestamp: 1648825140000,
+                    eventData: {},
+                    searchHub: 'in-product-help',
+                    document: {
+                      title: 'title',
+                      uriHash: 'TtnKwc0Lo2GY9WAi',
+                    },
+                  },
+                ],
+              },
+              {
+                start: 1648822380000,
+                end: 1648822620000,
+                actions: [
+                  {
+                    actionType: 'CUSTOM',
+                    timestamp: 1648822620000,
+                    eventData: {
+                      type: 'useless_event',
+                      value: '',
+                    },
+                    searchHub: 'community-support',
+                    document: {},
+                  },
+                  {
+                    actionType: 'CLICK',
+                    timestamp: 1648822500000,
+                    eventData: {},
+                    searchHub: 'in-product-help',
+                    document: {
+                      title: 'title',
+                      uriHash: 'KXñi9EWk38wnb1tt',
+                    },
+                  },
+                  {
+                    actionType: 'SEARCH',
+                    timestamp: 1648822380000,
+                    eventData: {},
+                    searchHub: 'in-product-help',
+                    document: {},
+                    query: '',
+                  },
+                ],
+              },
+            ],
+          };
+
+          const mappedAndSortedActions =
+            mapAndSortActionsByMostRecent(mockRawActions);
+          const ticketCreationDate = caseCreationDate.getTime();
+
+          const timeline = splitActionsIntoTimelineSessions(
+            mappedAndSortedActions,
+            ticketCreationDate
+          );
+          console.log(JSON.stringify(timeline));
+          expect(timeline).toEqual(expectedTimeline);
```

Comment:
I _think_ you should be using `to.deep.equal` instead here? since you want to compare the values inside the objects and not the object references themselves?

Please test with `deep.equal` instead when you have complex objects, `toEqual` is more for simple compare of values like this one you have too:

```
expect(timeline.precedingSessions.length).toEqual(
            expectedTimeline.precedingSessions.length
          );
          ```

Reply:
Reference:
https://www.chaijs.com/api/bdd/#method_deep

https://stackoverflow.com/a/63249954/8055339

Reply:
`to.deep.equal` is a `chai` assertion, here we are using Jest assertions, and the deep equality assertion with jest is made with `toEqual(...)`

Reply:
yea 🤦 across all the reviews I got confused haha. Thanks @mmitiche 

Ignore this comment.